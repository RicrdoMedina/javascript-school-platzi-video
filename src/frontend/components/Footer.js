import React from 'react';
import '../assets/styles/components/footer.scss';

const Footer = () => {
  return (
    <footer className='footer footer-dark'>
      <a href='/'>Terminos de uso</a>
      <a href='/'>Declaracion de privacidad</a>
      <a href='/'>Centro de ayuda</a>
    </footer>
  );
};

export default Footer;
