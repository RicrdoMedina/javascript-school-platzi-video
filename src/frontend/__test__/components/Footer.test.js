import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Footer from '../../components/Footer';

describe('<Footer/>', () => {
  //A la suit le colocamos el nombre del componente

  const footer = mount(<Footer />); // Montamos nuestro componentes.

  test('Render Footer Component', () => {
    //Probar que el componente haga render
    expect(footer.length).toEqual(1);
  });

  test('Footer haves 3 anchors', () => {
    //Probar que tenga 3 elementos anchor
    expect(footer.find('a')).toHaveLength(3);
  });

  test('Footer Snapshot', () => {
    const footer = create(<Footer />); // Creamos una nueva constante footer que solo trabaje en este scope

    expect(footer.toJSON()).toMatchSnapshot(); //Creamos el snapshot. Si no esta creado lo crea y la siguiente vez compara el snapshot con el componente
  });
});
