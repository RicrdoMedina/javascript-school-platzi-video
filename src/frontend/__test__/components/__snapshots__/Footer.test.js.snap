// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> Footer Snapshot 1`] = `
<footer
  className="footer footer-dark"
>
  <a
    href="/"
  >
    Terminos de uso
  </a>
  <a
    href="/"
  >
    Declaracion de privacidad
  </a>
  <a
    href="/"
  >
    Centro de ayuda
  </a>
</footer>
`;
